### PR TITLE
fix 2d/3d if not in xy plane

### PR DESCRIPTION
**Pull request by jh66637**
Thanks for adding CGAL support. I am very interested in this :)

I tested the intersections and encountered problems for 2d/3d intersections. This happens only if the 2d element is not in the (x,y) plane. Using also a Triangulation3_exact for the 2d elements and iterating over its faces cures this issue.

I am currently rebuilding and will check if this is the same case for other interdimensional intersections, but I need some time to rebuild and test everything :)

### CODE REVIEW

**Comment by jh66637** [comment]
I do not see similar problems for 1d/3d and 1d/2d intersections, so I think the PR would be ready from my side.
@fdrmrc: Do you agree to the changes?

**Comment by fdrmrc** [comment]
Thanks for testing all of this. Much appreciated! :) 

What was the problem you were observing? Do you still have a MWE for that? 
For what concerns 2D/3D intersections, I also have to say I found a bug in `CGAL::intersection()` that has been fixed but not yet merged. (https://github.com/CGAL/cgal/issues/6777)

**Comment by jh66637** [comment]
The problem I had was that CGAL only detects an intersection between a tetrahedron and a triangle if the triangle is defined in the (x,y) plane. As far as I can tell, this is the case because Delaunay_triangulation_2 takes 2D Points, but the triangle can be oriented arbitrarily in the 3D space. So I think this is not related to the mentioned PR.

Regarding the MWE: I adapted the test such that the 2D triangulation is defined int the (x,y) (x,z) and (y,z) plane. Without the modifications in intersection.cc, only the first triangulation succeeds.

File: source/cgal/intersections.cc [hunk 1]
```diff
@@ -0,0 +1,739 @@
...
+using CGALSegment3           = K::Segment_3;
+using CGALSegment3_exact     = K_exact::Segment_3;
+using CGALTetra              = K::Tetrahedron_3;
+using CGALTetra_exact        = K_exact::Tetrahedron_3;
+using Triangulation2         = CGAL::Triangulation_2<K>;
+using Triangulation3         = CGAL::Triangulation_3<K>;
+using Triangulation3_exact   = CGAL::Triangulation_3<K_exact>;
+using Triangulation3_inexact = CGAL::Triangulation_3<K_inexact>;
+typedef CGAL::Projection_traits_xy_3<K_exact> Gt;
+typedef CGAL::Delaunay_triangulation_2<Gt>    Delaunay;
```
Comment (fdrmrc) [inline source/cgal/intersections.cc:81]: These two typedefs (line 80 and 81) can be removed now

File: tests/cgal/cgal_intersect_simplices_2d_3d.cc [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+  const unsigned int j;
+};
+
+void
+test_intersection_inside(Triangulation<3> &tria0, Triangulation<2, 3> &tria1)
+{
+  GridGenerator::hyper_cube(tria0, -1., 1.);
+  GridGenerator::hyper_cube(tria1, -0.4, 0.2);
+
+  // GridOut().write_mesh_per_processor_as_vtu(tria0, "Tria3d");
```
Comment (fdrmrc) [inline tests/cgal/cgal_intersect_simplices_2d_3d.cc:56]: Remove this and the next calls to `write_mesh_per_processor_as_vtu()`

**Review by fdrmrc — APPROVED** [review]
Thanks a lot. I rebased my PR on top of dealii master. Once the conflicts are fixed I'll merge it

**Pull request merged**